lib/dao.js

```diff
@@ -230,7 +231,9 @@ DataAccessObject.create = function (data, options, cb) {
   Model = this.lookupModel(data); // data-specific
   if (Model !== obj.constructor) obj = new Model(data);
 
-  Model.notifyObserversOf('before save', { Model: Model, instance: obj }, function(err) {
+  Model.notifyObserversOf('before save', {
+    Model: Model, instance: obj, hookState: hookState
+  }, function(err) {
```

Comment:
coding style: please move the context initialisation to a new line and property.

```
var context = ...;
Model.notifyObserversOf('before save', context, function(err) {
```

The same fix should be applied to all other places where the context initialization does not fit a single line.

IIRC, we are already using the style described above in other existing code, so let's keep the style consistent.